sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/CollectionConfiguration.cs

```diff
@@ -0,0 +1,225 @@
+    /// </remarks>
+    internal class CollectionConfiguration
+    {
+        private readonly CollectionConfigurationInfo info;
+
+        #region Collection-time instances used to filter and calculate data on telemetry passing through the pipeline
+        private readonly List<DerivedMetric<Request>> requestDocumentIngressMetrics = new List<DerivedMetric<Request>>();
+
+        private readonly List<DerivedMetric<RemoteDependency>> dependencyTelemetryMetrics =
+            new List<DerivedMetric<RemoteDependency>>();
+
+        private readonly List<DerivedMetric<ExceptionDocument>> exceptionTelemetryMetrics =
+            new List<DerivedMetric<ExceptionDocument>>();
+
+        // private readonly List<DerivedMetric<EventTelemetry>> eventTelemetryMetrics = new List<DerivedMetric<EventTelemetry>>();
+
+        // private readonly List<DerivedMetric<TraceTelemetry>> traceTelemetryMetrics = new List<DerivedMetric<TraceTelemetry>>();
+
+        // TODO: Add back: private readonly List<DocumentStream> documentStreams = new List<DocumentStream>();
+        #endregion
+
+        #region Metadata used by other components
+        private readonly List<Tuple<string, DerivedMetricInfoAggregation?>> telemetryMetadata = new List<Tuple<string, DerivedMetricInfoAggregation?>>();
+
+        private readonly List<Tuple<string, string>> performanceCounters = new List<Tuple<string, string>>();
+        #endregion
+
+        public CollectionConfiguration(
+           CollectionConfigurationInfo info,
+           out CollectionConfigurationError[] errors)
+        {
+            this.info = info ?? throw new ArgumentNullException(nameof(info));
+
+            // create metrics based on descriptions in info
+            this.CreateTelemetryMetrics(out CollectionConfigurationError[] metricErrors);
+
+            // maintain a separate collection of all (Id, AggregationType) pairs with some additional data - to allow for uniform access to all types of metrics
+            // this includes both telemetry metrics and Metric metrics
+            this.CreateMetadata();
+
+            //// create document streams based on description in info
+            //this.CreateDocumentStreams(out CollectionConfigurationError[] documentStreamErrors, timeProvider, previousDocumentStreams ?? ArrayExtensions.Empty<DocumentStream>());
+
+            //// create performance counters
+            //this.CreatePerformanceCounters(out CollectionConfigurationError[] performanceCounterErrors);
+
+            //errors = metricErrors.Concat(documentStreamErrors).Concat(performanceCounterErrors).ToArray();
+            errors = metricErrors.ToArray();
+
+            foreach (var error in errors)
+            {
+                UpdateMetricIdOfError(error, this.info.Etag);
+            }
+        }
+
+        private void UpdateMetricIdOfError(CollectionConfigurationError error, string id)
+        {
+            for (int i = 0; i < error.Data.Count; i++)
```

Comment:
I see a slight logic change, could you explain?

Reply:
`CollectionConfigurationError.Data` was of type `Dictionary<string, string>` but now is of type `IList<KeyValuePairString>`. Not sure why this change is made but it has made updating value for a key much more hassle.